### ISSUE TITLE
man page not providing any details

### ISSUE THREAD

**Issue by cadm-frank** [NixOS/nix#6625]
**Describe the bug**

"man nix" only lists the commands, but does not provide any useful hints on how to actually use nix:
```
       Main commands:

       • nix build <./nix3-build.md> - build a derivation or fetch a store path
       • nix develop <./nix3-develop.md> - run a bash shell that provides the build environment of a derivation
       • nix flake <./nix3-flake.md> - manage Nix flakes
[...]
       • nix shell <./nix3-shell.md> - run a shell in which the specified packages are available
```

**Steps To Reproduce**

1. Install and run NixOS 22.05.583.ca2629644d4
2. `$ man nix`

**Expected behavior**

I would have expected at least a hint on where to find additional information about each of those commands, if not in this man page, at least a hint of the correct man page. How do I use e.g. "nix shell"?

**`nix-env --version` output**
```
$ nix-env --version
nix-env (Nix) 2.8.1
```

**Additional context**

**Comment by thufschmitt** [comment]
Below this list of commands, there should be some examples and a more detailed description of Nix in general. Are these missing from your manpage or do you mean that it's hard to discover because it's burried under a lot of less useful stuff?

**Comment by dfbm** [comment]
@thufschmitt Take "nix run". What does it do?
The man page gives you the following information:
- run a Nix application
- Run the flake in the current directory with "nix run"
That's it. No further information available.

What is a Nix application?
What if I want to run something besides the flake in the current directory? Does it support any arguments?
How do I specify which "Nix application" (whatever that is) to run? What are the arguments that "nix run" specifically supports if I want to run e.g. VLC from the current nixpkgs channel on the system?

"nix search" - 'search for packages'
Ok. How can I search? What's the scope of the search? The current default channel or all channels? Can I change the scope of the search to only search names and/or description?

Each of these commands would probably warrant enough information to fill an additional man page. Are there additional man pages? What are their names? What section are they in?

**Comment by thufschmitt** [comment]
> Each of these commands would probably warrant enough information to fill an additional man page. Are there additional man pages? What are their names? What section are they in?

Yes, that's an issue with the manpages, the source uses hyperlinks to point to the right pages, but they don't render properly in man – the renderer just dumps the link as `<./nix3-build.md>`, which is inefficient. As Eelco said, the doc is available with `--help` (or as a  manpage with `man nix3-{subcommand}`). Not sure how to convey that information properly in the toplevel help

**Comment by tomberek** [comment]
Closing as stale. Re-open if needed.
